1. Задачка с CoreDNS - есть ошибка в манифестах (смотрите документацию по ссылке в ДЗ): 
```____________________________ test_coredns_shared_ip ____________________________
tests/test_220_coredns.py:140: in test_coredns_shared_ip
AssertionError: UDP (172.17.255.2) and TCP(172.17.255.1) LB IPs don't match
```
2. Ingress для Dashboard -  рабочий, все нормально
3. Повесить dashboard и web в канареечный релиз - оригинально, мне понравилось. Манифесты тоже рабочие.

Если хотите исправить недочеты - отпишитесь. Если не хотите - я могу принять ДЗ в таком виде.

--- a/kubernetes-networks/README.md
+++ b/kubernetes-networks/README.md
@@ -41,10 +41,11 @@ nginx/master/deploy/static/mandatory.yaml) и добавлена [конфигу
 Задание:
 *Создал [Ingress правило](./dashboard-ingress.yaml) для доступа к Dashboard*
 
-
+Создал манифест для [канареечного тестирования](./canary-ingress.yaml)
 
 ---
 Полезные ссылки:
+- [MetalLB Usage](https://metallb.universe.tf/usage/)
 - [Ingress Nginx rewrite](https://kubernetes.github.io/ingress-nginx/examples/rewrite/)
 - [Canary ingress](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#canary)
 

--- a/kubernetes-networks/dns-svc-lb.yaml
+++ b/kubernetes-networks/dns-svc-lb.yaml
@@ -3,6 +3,8 @@ kind: Service
 metadata:
   name: dns-udp-svc-lb
   namespace: kube-system
+  annotations:
+    metallb.universe.tf/allow-shared-ip: "coredns"
 spec:
   selector:
     k8s-app: kube-dns
@@ -17,6 +19,8 @@ kind: Service
 metadata:
   name: dns-tcp-svc-lb
   namespace: kube-system
+  annotations:
+    metallb.universe.tf/allow-shared-ip: "coredns"
 spec:
   selector:
     k8s-app: kube-dns

--- a/kubernetes-networks/metallb-config.yaml
+++ b/kubernetes-networks/metallb-config.yaml
@@ -9,5 +9,5 @@ data:
     - name: default
       protocol: layer2
       addresses:
-      - "172.17.255.1-172.17.255.254"
+      - "172.17.255.1-172.17.255.255"
       

